### PR TITLE
temperature: don't enable for monochrome raws

### PR DESCRIPTION
**Pull request by kelvie**
This does two things -- it prevents an error message in prepare_matrices (when
the colour matrix is invariably not found for the monochorme raw), and it does
not enable the white balance module by default for monochrome raws.

This moves the is_leica_monochrom check outside the is_raw block.

### CODE REVIEW

File: src/iop/temperature.c [hunk 1]
```diff
@@ -938,10 +938,16 @@ void reload_defaults(dt_iop_module_t *module)
   // we might be called from presets update infrastructure => there is no image
   if(!module->dev) goto end;
 
+  int is_raw = dt_image_is_raw(&module->dev->image_storage);
+
+  // White balance module doesn't need to be anbled for monochrome raws (like
+  // for leica monochrom cameras)
+  if(is_raw && is_leica_monochrom(&(module->dev->image_storage))) goto gui;
+
   if(module->gui_data) prepare_matrices(module);
```
Comment (LebedevRI) [inline src/iop/temperature.c]: So what happens if this is not called?

File: src/iop/temperature.c [hunk 2]
```diff
@@ -938,10 +938,16 @@ void reload_defaults(dt_iop_module_t *module)
   // we might be called from presets update infrastructure => there is no image
   if(!module->dev) goto end;
 
+  int is_raw = dt_image_is_raw(&module->dev->image_storage);
+
+  // White balance module doesn't need to be anbled for monochrome raws (like
```
Comment (LebedevRI) [inline src/iop/temperature.c]: Typo

File: src/iop/temperature.c [hunk 3]
```diff
@@ -938,10 +938,16 @@ void reload_defaults(dt_iop_module_t *module)
   // we might be called from presets update infrastructure => there is no image
   if(!module->dev) goto end;
 
+  int is_raw = dt_image_is_raw(&module->dev->image_storage);
```
Comment (LebedevRI) [inline src/iop/temperature.c]: `const`

File: src/iop/temperature.c (hunk 1 above)
Comment (kelvie) [reply src/iop/temperature.c]: Did you mean if `reload_defaults` isn't called, or if `prepare_matrices` isn't called?

Monochrom raws don't have colour matrices anyway,  so all `prepare_matrices` does for monochrom raws is write an error messas.

I'll fixup the other stuff.

Comment (LebedevRI) [reply src/iop/temperature.c]: If prepare_matrices isn't called.

> Monochrom raws don't have colour matrices anyway, so all prepare_matrices does for monochrom raws is write an error messas.

Ah, now that i checked, the temperature/tint slider is already non-functional.

**Comment by kelvie** [comment]
FYI I fixed the comments above (can't see my replies to the inline comments anymore it seems, when I fixup and re-push)

**Comment by kelvie** [comment]
Going to close this and put in all the other monochrom fixes in the same PR...

**Pull request closed without merging**